signac/core/h5store.py

```diff
@@ -55,12 +55,22 @@ def _requires_tables():
 logger = logging.getLogger(__name__)
 
 
+class ClosedFileError(RuntimeError):
+    "Raised when trying to access a closed group."
+    pass
+
+
 def _h5set(file, grp, key, value, path=None):
     """Set a key in an h5py container, recursively converting Mappings to h5py
     groups and transparently handling None."""
+    import h5py
     import numpy    # h5py depends on numpy, so this is safe.
     path = path + '/' + key if path else key
 
+    if isinstance(value, H5Group) and key in grp:
+        if grp[key] == value._group:
```

Comment:
Is this check for performance or correctness? If it's correct without it, I'm not sure if we should do this check.

Reply:
This is required for correctness. Otherwise you delete the key and then try to assign the value that was just deleted to the same key.

Reply:
A comment might be helpful. e.g.
`# Guards against setting a key to itself, which can result in deletion before assignment`